tests/test_trainer.py

```diff
@@ -81,6 +81,23 @@ def test_trainer_raises_error_with_missing_text(self):
         with pytest.raises(ValueError):
             trainer.train()
 
+    def test_trainer_raises_warning_with_missing_text(self):
+        """If the required columns are missing from the dataset, the library should log a warning and list the columns found."""
+        dataset = Dataset.from_dict({"label": [0, 1, 2], "extra_column": ["d", "e", "f"]})
+        trainer = SetFitTrainer(
+            model=self.model, train_dataset=dataset, eval_dataset=dataset, num_iterations=self.num_iterations
+        )
+        with self.assertLogs(level=logging.WARNING) as cm:
+            try:
+                trainer._validate_column_mapping(trainer.train_dataset)
+            except ValueError:
+                pass
```

Comment:
```suggestion
            with self.assertRaises(ValueError):
                trainer._validate_column_mapping(trainer.train_dataset)
```
Perhaps we can adopt `assertRaises` here?